packaging/packaging.go

```diff
@@ -0,0 +1,293 @@
+	// write the manifest file
+	file, _ := json.MarshalIndent(fileManifest, "", " ")
+	_ = ioutil.WriteFile(manifestFileName, file, 0644)
+	// return the manifest file/uuid
+	log.Info("Generated manifest file", "manifest", manifestFileName)
+	return manifestFileName, manifestUUID
+}
+
+func addFileToTarWriter(logger logr.Logger, uploadName, filePath string, tarWriter *tar.Writer) error {
+	log := logger.WithValues("costmanagement", "addFileToTarWriter")
+	log.Info("Adding file to tar.gz", "file", filePath)
+	file, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	stat, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	header := &tar.Header{
+		Name:    uploadName,
+		Size:    stat.Size(),
+		Mode:    int64(stat.Mode()),
+		ModTime: stat.ModTime(),
+	}
+
+	err = tarWriter.WriteHeader(header)
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(tarWriter, file)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func WriteTarball(logger logr.Logger, tarFileName, manifestFileName, manifestUUID string, archiveFiles []string) string {
+	if len(archiveFiles) > 0 {
+		// create the tarfile
+		tarFile, err := os.Create(tarFileName)
+		if err != nil {
+			fmt.Println("Error!")
+		}
+		defer tarFile.Close()
+
+		gzipWriter := gzip.NewWriter(tarFile)
+		defer gzipWriter.Close()
+
+		tw := tar.NewWriter(gzipWriter)
+		defer tw.Close()
+
+		// add the files to the tarFile
+		for idx, fileName := range archiveFiles {
+			fmt.Println(fileName)
+			if strings.Contains(fileName, ".csv") {
+				uploadName := manifestUUID + "_openshift_usage_report." + strconv.Itoa(idx) + ".csv"
+				fmt.Println(uploadName)
+				err := addFileToTarWriter(logger, uploadName, fileName, tw)
+				if err != nil {
+					fmt.Println(err)
+					return ""
+				}
+			}
+		}
+		addFileToTarWriter(logger, "manifest.json", manifestFileName, tw)
+
+		return tarFileName
+
+	}
+	return ""
+}
+
+func WritePart(logger logr.Logger, fileName string, csvReader *csv.Reader, num int) (string, bool) {
+	log := logger.WithValues("costmanagement", "WritePart")
+	fileNamePart := strings.TrimSuffix(fileName, ".csv")
+	sizeEstimate := 0
+	splitFileName := fileNamePart + strconv.Itoa(num) + ".csv"
+	log.Info("Creating file ", "file", splitFileName)
+	splitFile, err := os.Create(splitFileName)
+	if err != nil {
+		fmt.Println("An error occurred:", err)
+	}
+	// Create the csv writer
+	writer := csv.NewWriter(splitFile)
+	// err = writer.WriteRow(csvHeader)
+	for {
+		row, err := csvReader.Read()
+		if err == io.EOF {
+			return splitFileName, true
+		}
+		writer.Write(row)
+		rowLen := len(strings.Join(row, ","))
+		rowSize := rowLen + int(float64(rowLen)*variance)
+		sizeEstimate += rowSize
+		if sizeEstimate >= int(maxBytes) {
+			return splitFileName, false
+		}
+	}
+}
+
+func SplitFiles(logger logr.Logger, filePath string) {
+	fileList, err := ioutil.ReadDir(filePath)
+	if err != nil {
+		fmt.Println("could not read dir")
```

Comment:
What happens when we can't read the directory?

Reply:
This error is now bubbled up to the reconciler and we do not proceed if we can't read the directory.